zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/RoleTest.java

```diff
@@ -52,4 +54,81 @@ public void shouldNotDuplicate() {
                 + name
                 + "', but a role with this name already exists");
   }
+
+  @Test
+  public void shouldAddEntityToRole() {
+    final var userKey =
+        ENGINE
+            .user()
+            .newUser("username")
+            .withName("Foo Bar")
+            .withEmail("foo@bar.com")
+            .withPassword("password")
+            .create()
+            .getValue()
+            .getUserKey();
+    final var name = UUID.randomUUID().toString();
+    final var roleKey = ENGINE.role().newRole(name).create().getValue().getRoleKey();
+    final var updatedRole =
+        ENGINE
+            .role()
+            .addEntity(roleKey)
+            .withEntityKey(userKey)
+            .withEntityType(EntityType.USER)
+            .add()
+            .getValue();
+
+    Assertions.assertThat(updatedRole)
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("entityKey", userKey);
+  }
+
+  @Test
+  public void shouldRejectIfRoleIsNotPresent() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var roleRecord = ENGINE.role().newRole(name).create();
+
+    // when
+    final var notPresentRoleKey = 1L;
+    final var notPresentUpdateRecord =
+        ENGINE.role().addEntity(notPresentRoleKey).expectRejection().add();
+
+    final var createdRole = roleRecord.getValue();
+    Assertions.assertThat(createdRole).isNotNull().hasFieldOrPropertyWithValue("name", name);
+
+    assertThat(notPresentUpdateRecord)
+        .hasRejectionType(RejectionType.NOT_FOUND)
+        .hasRejectionReason(
+            "Expected to update role with key '"
+                + notPresentRoleKey
+                + "', but a role with this key does not exists");
+  }
+
+  @Test
+  public void shouldRejectIfEntityIsNotPresent() {
```

Comment:
❓ Similar to my question with the "role update" PR: does it make sense to add a test for rejection if unauthorized?

Or would that go in the authorization unit, or IT tests?

Reply:
We have IT tests to verify this behaviour of the processors (see [here](https://github.com/camunda/camunda/tree/main/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/authorization)). Since I'm working on refactor them and the behaviour class is tested, I'd say is out of scope for this PR